From 075c6f239e4f3384bac840c5efbcfed0b7024f1c Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Mon, 26 Aug 2024 22:32:36 +0200
Subject: [PATCH 1/3] Fix missing headers
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

/run/build/BambuStudio/src/libslic3r/PerimeterGenerator.cpp: In function ‘double Slic3r::random_value()’:
/run/build/BambuStudio/src/libslic3r/PerimeterGenerator.cpp:31:76: error: ‘thread’ is not a member of ‘std’
   31 |     thread_local std::mt19937 gen(rd.entropy() > 0 ? rd() : std::hash<std::thread::id>()(std::this_thread::get_id()));
      |                                                                            ^~~~~~
/run/build/BambuStudio/src/libslic3r/PerimeterGenerator.cpp:15:1: note: ‘std::thread’ is defined in header ‘<thread>’; did you forget to ‘#include <thread>’?
   14 | #include "libslic3r/AABBTreeLines.hpp"
  +++ |+#include <thread>
   15 | static const int overhang_sampling_number = 6;
---
 src/libslic3r/PerimeterGenerator.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libslic3r/PerimeterGenerator.cpp b/src/libslic3r/PerimeterGenerator.cpp
index 0a2cc8a240d6..f992cb640faf 100644
--- a/src/libslic3r/PerimeterGenerator.cpp
+++ b/src/libslic3r/PerimeterGenerator.cpp
@@ -10,6 +10,7 @@
 #include <cmath>
 #include <cassert>
 #include <random>
+#include <thread>
 #include <unordered_set>
 #include "libslic3r/AABBTreeLines.hpp"
 static const int overhang_sampling_number = 6;
-- 
2.45.2


From 66ef60ded016c8f5071f5c272513743e7079a1e3 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 27 Aug 2024 09:20:53 +0200
Subject: [PATCH 2/3] Fix missing BOOST_LOG_TRIVIAL declaration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

/run/build/BambuStudio/src/libslic3r/Flow.cpp: In static member function ‘static float Slic3r::Flow::rounded_rectangle_extrusion_spacing(float, float)’:
/run/build/BambuStudio/src/libslic3r/Flow.cpp:194:27: error: ‘error’ was not declared in this scope; did you mean ‘nlohmann::detail::cbor_tag_handler_t::error’?
  194 |         BOOST_LOG_TRIVIAL(error)<< __FUNCTION__ << boost::format("negative extrusion : width %1%   height %2%") % width % height;
      |                           ^~~~~
      |                           nlohmann::detail::cbor_tag_handler_t::error
---
 src/libslic3r/Flow.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libslic3r/Flow.cpp b/src/libslic3r/Flow.cpp
index 97b3c1427462..b7edb0f558b0 100644
--- a/src/libslic3r/Flow.cpp
+++ b/src/libslic3r/Flow.cpp
@@ -5,6 +5,7 @@
 #include <assert.h>
 
 #include <boost/algorithm/string/predicate.hpp>
+#include <boost/log/trivial.hpp>
 
 // Mark string for localization and translate.
 #define L(s) Slic3r::I18N::translate(s)
-- 
2.45.2


From 1b7b1f16bbbaf586f1101f52ba178851c07f29b8 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 27 Aug 2024 09:54:13 +0200
Subject: [PATCH 3/3] Add missing regex declaration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

/run/build/BambuStudio/src/slic3r/GUI/WebViewDialog.cpp: In member function ‘void Slic3r::GUI::WebViewPanel::SetMakerworldPageLoginStatus(bool, wxString)’:
/run/build/BambuStudio/src/slic3r/GUI/WebViewDialog.cpp:818:10: error: ‘regex’ is not a member of ‘std’; did you mean ‘boost::regex’?
  818 |     std::regex pattern("^https://.*/(.*/){0,1}agree-terms.*");
      |          ^~~~~
---
 src/slic3r/GUI/WebViewDialog.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/slic3r/GUI/WebViewDialog.cpp b/src/slic3r/GUI/WebViewDialog.cpp
index ea4d4d0dcee8..7a2d9225fb10 100644
--- a/src/slic3r/GUI/WebViewDialog.cpp
+++ b/src/slic3r/GUI/WebViewDialog.cpp
@@ -7,6 +7,8 @@
 #include "libslic3r_version.h"
 #include "../Utils/Http.hpp"
 
+#include <regex>
+
 #include <boost/property_tree/ptree.hpp>
 #include <boost/property_tree/json_parser.hpp>
 
-- 
2.45.2

